x-pack/plugins/observability/public/pages/threshold/components/expression.tsx

```diff
@@ -0,0 +1,500 @@
+import { ExpressionRow } from './expression_row';
+import { MetricsExplorerKueryBar } from './kuery_bar';
+import { MetricsExplorerOptions } from '../hooks/use_metrics_explorer_options';
+import { convertKueryToElasticSearchQuery } from '../helpers/kuery';
+import { useSourceContext, withSourceProvider } from '../helpers/source';
+import { MetricsExplorerGroupBy } from './group_by';
+const FILTER_TYPING_DEBOUNCE_MS = 500;
+
+type Props = Omit<
+  RuleTypeParamsExpressionProps<RuleTypeParams & AlertParams, AlertContextMeta>,
+  'defaultActionGroupId' | 'actionGroups' | 'charts' | 'data' | 'unifiedSearch' | 'onChangeMetaData'
+>;
+
+const defaultExpression = {
+  aggType: Aggregators.AVERAGE,
+  comparator: Comparator.GT,
+  threshold: [],
+  timeSize: 1,
+  timeUnit: 'm',
+} as MetricExpression;
+export { defaultExpression };
+
+export function Expressions(props: Props) {
+  const { setRuleParams, ruleParams, errors, metadata } = props;
+  const { docLinks } = useKibana().services;
+  const { source, createDerivedIndexPattern } = useSourceContext();
+
+  const [timeSize, setTimeSize] = useState<number | undefined>(1);
+  const [timeUnit, setTimeUnit] = useState<TimeUnitChar | undefined>('m');
+  const derivedIndexPattern = useMemo(
+    () => createDerivedIndexPattern(),
+    [createDerivedIndexPattern]
+  );
+
+  const options = useMemo<MetricsExplorerOptions>(() => {
+    if (metadata?.currentOptions?.metrics) {
+      return metadata.currentOptions as MetricsExplorerOptions;
+    } else {
+      return {
+        metrics: [],
+        aggregation: 'avg',
+      };
+    }
+  }, [metadata]);
+
+  const updateParams = useCallback(
+    (id, e: MetricExpression) => {
+      const exp = ruleParams.criteria ? ruleParams.criteria.slice() : [];
+      exp[id] = e;
+      setRuleParams('criteria', exp);
+    },
+    [setRuleParams, ruleParams.criteria]
+  );
+
+  const addExpression = useCallback(() => {
+    const exp = ruleParams.criteria?.slice() || [];
+    exp.push({
+      ...defaultExpression,
+      timeSize: timeSize ?? defaultExpression.timeSize,
+      timeUnit: timeUnit ?? defaultExpression.timeUnit,
+    });
+    setRuleParams('criteria', exp);
+  }, [setRuleParams, ruleParams.criteria, timeSize, timeUnit]);
+
+  const removeExpression = useCallback(
+    (id: number) => {
+      const exp = ruleParams.criteria?.slice() || [];
+      if (exp.length > 1) {
+        exp.splice(id, 1);
+        setRuleParams('criteria', exp);
+      }
+    },
+    [setRuleParams, ruleParams.criteria]
+  );
+
+  const onFilterChange = useCallback(
+    (filter: any) => {
+      setRuleParams('filterQueryText', filter);
+      try {
+        setRuleParams(
+          'filterQuery',
+          convertKueryToElasticSearchQuery(filter, derivedIndexPattern, false) || ''
+        );
+      } catch (e) {
+        setRuleParams('filterQuery', QUERY_INVALID);
+      }
+    },
+    [setRuleParams, derivedIndexPattern]
+  );
```

Comment:
I would like this changed so we only store the KQL expression and not the compiled JSON. Currently `fitlerQueryText` is the unparsed KQL expression and `filterQuery` is a stringified JSON representation of the KQL expression. This also makes the JSON body for creating a rule via the API confusing.  We should only compile the KQL expression on the server.